### PR TITLE
[Task] Create csv export for folder with grid config

### LINKED ISSUE #472: [Task] Add endpoint to CSV-Download to download whole folder
See #440

### PR DESCRIPTION
**Pull request by mattamon**
## Changes in this pull request
Resolves #472 

## Additional info
ADd endpoint to export assets based on a folder. Additional give possibility to use gridfilters

### CODE REVIEW

File: src/Asset/Controller/Export/CsvFolderController.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    #[CsvExportFolderRequestBody]
+    #[CreatedResponse(
+        description: 'asset_export_csv_created_response',
+        content: new IdJson('ID of created jobRun', 'jobRunId')
+    )]
+    #[DefaultResponses([
+        HttpResponseCodes::UNAUTHORIZED,
+        HttpResponseCodes::NOT_FOUND,
+    ])]
+    public function assetCreateCsv(
```
Comment (martineiber) [inline src/Asset/Controller/Export/CsvFolderController.php]:
```suggestion
    public function assetExportCsvFolder(
```

File: src/Asset/Controller/Export/CsvFolderController.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+final class CsvFolderController extends AbstractApiController
+{
+    public function __construct(
+        SerializerInterface $serializer,
+        private readonly CsvServiceInterface $csvService
+    ) {
+        parent::__construct($serializer);
+    }
+
+    #[Route('/assets/export/csv/folder', name: 'pimcore_studio_api_export_csv_folder', methods: ['POST'])]
```
Comment (martineiber) [inline src/Asset/Controller/Export/CsvFolderController.php]: I would say we should add asset to the name?

```suggestion
    #[Route('/assets/export/csv/folder', name: 'pimcore_studio_api_asset_export_csv_folder', methods: ['POST'])]
```

**Review by martineiber — CHANGES_REQUESTED** [review]

File: src/Asset/Controller/Export/CsvFolderController.php (hunk 2 above)
Comment (mattamon) [reply src/Asset/Controller/Export/CsvFolderController.php]: You are totally right :D was too much to change already! Thanks for noticing it!

**Review by martineiber — APPROVED** [review]
LGTM :seal:

**Pull request merged**